src/super_gradients/training/models/model_factory.py

```diff
@@ -0,0 +1,102 @@
+from typing import Union
+
+import torch
+import torch.nn as nn
+
+from super_gradients.training import utils as core_utils
+from super_gradients.training.models import SgModule
+from super_gradients.training.models.all_architectures import ARCHITECTURES
+from super_gradients.training.pretrained_models import PRETRAINED_NUM_CLASSES
+from super_gradients.training.utils.checkpoint_utils import load_checkpoint_to_model, load_pretrained_weights, \
+    read_ckpt_state_dict
+
+
+class SgNetsFactory:
+    @classmethod
+    def get(cls, architecture: Union[str, nn.Module], arch_params={}, checkpoint_params={}, *args,
```

Comment:
please define type for arch_params and checkpoint_params

Reply:
added